In [3]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt


In [4]:
df = df = pd.read_excel('/content/drive/MyDrive/Advanced_Analytics_using_statistics/CDAC_DataBook.xlsx' ,sheet_name='faithful')

In [5]:
df.head()

,eruptions,waiting
0,3.600,79
1,1.800,54
2,3.333,74
3,2.283,62
4,4.533,85


In [6]:
x1 = df.eruptions

In [7]:
print(x1)

0      3.600
1      1.800
2      3.333
3      2.283
4      4.533
       ...  
267    4.117
268    2.150
269    4.417
270    1.817
271    4.467
Name: eruptions, Length: 272, dtype: float64


In [8]:
np.mean(x1)

np.float64(3.4877830882352936)

In [9]:
len(x1)

272

In [10]:
sum(x1)/len(x1)

3.487783088235294

Importing the remaining library in it.


In [11]:
import scipy
from scipy import stats

In [12]:
stats.mode(x1)

ModeResult(mode=np.float64(1.867), count=np.int64(8))

In [13]:
len(np.where(x1==1.867)[0])

8

In [14]:
from collections import Counter

In [15]:
Counter(x1).most_common()

[(1.867, 8),
 (4.5, 8),
 (1.833, 7),
 (4.7, 6),
 (1.75, 6),
 (4.8, 6),
 (4.0, 6),
 (4.533, 5),
 (4.083, 5),
 (3.833, 5),
 (4.333, 5),
 (3.6, 4),
 (1.8, 4),
 (4.35, 4),
 (4.25, 4),
 (1.883, 4),
 (2.0, 4),
 (4.167, 4),
 (4.417, 4),
 (4.6, 4),
 (4.15, 4),
 (4.583, 4),
 (3.917, 3),
 (1.967, 3),
 (2.017, 3),
 (4.567, 3),
 (2.1, 3),
 (4.633, 3),
 (1.817, 3),
 (1.983, 3),
 (4.933, 3),
 (2.2, 3),
 (4.367, 3),
 (4.233, 3),
 (4.45, 3),
 (3.333, 2),
 (2.167, 2),
 (3.85, 2),
 (4.433, 2),
 (4.3, 2),
 (4.467, 2),
 (4.033, 2),
 (4.833, 2),
 (2.233, 2),
 (4.133, 2),
 (4.1, 2),
 (4.067, 2),
 (3.95, 2),
 (4.817, 2),
 (4.667, 2),
 (4.9, 2),
 (1.783, 2),
 (3.767, 2),
 (1.917, 2),
 (2.033, 2),
 (2.4, 2),
 (3.567, 2),
 (3.5, 2),
 (1.933, 2),
 (2.083, 2),
 (2.417, 2),
 (2.25, 2),
 (4.267, 2),
 (4.117, 2),
 (1.85, 2),
 (4.283, 2),
 (2.283, 1),
 (2.883, 1),
 (1.95, 1),
 (4.2, 1),
 (1.6, 1),
 (3.45, 1),
 (3.067, 1),
 (3.367, 1),
 (4.783, 1),
 (3.317, 1),
 (4.716, 1),
 (1.733, 1),
 (4.883, 1),
 (3.717, 1),
 (1.6

In [16]:
#

max(x1) - min(x1)

3.4999999999999996

In [17]:
np.ptp(x1)

np.float64(3.4999999999999996)

In [18]:
# IQR formula
IQR = np.percentile(x1,75) - np.percentile(x1,25)
IQR

np.float64(2.2915)

In [19]:
np.var(x1)

1.2979388904492855

In [20]:
np.std(x1)

1.1392712102257678

In [21]:
np.var(x1, ddof=1)

1.3027283328494672

In [22]:
np.std(x1,ddof=1)

1.1413712511052077

In [29]:
np.std(x1)

1.1392712102257678

In [24]:
from scipy.stats import binom
'''
Que :
Chances of rain on any given day in June in Mumbai is 0.7
If I am in Mumbai for 15 days, what is the probability that
a. It rains for exactly 10 days
b. It rains for maximum 10 days
c. It rains for minimum 10 days

binom.pmf= probability mass function (to fund probability of Exact number)
binom.cdf = cummulative distribution function (to fund probability of upto a particular number)
binom.sf = survival function (to find probability of MORE THAN a particular number)
'''

' \nQue :\nChances of rain on any given day in June in Mumbai is 0.7\nIf I am in Mumbai for 15 days, what is the probability that\na. It rains for exactly 10 days\nb. It rains for maximum 10 days\nc. It rains for minimum 10 days\n\nbinom.pmf= probability mass function (to fund probability of Exact number)\nbinom.cdf = cummulative distribution function (to fund probability of upto a particular number)\nbinom.sf = survival function (to find probability of MORE THAN a particular number)\n'

In [25]:
binom.pmf(10,15,0.7)

np.float64(0.20613038097752098)

In [26]:
binom.cdf(10,15,0.7)

np.float64(0.4845089407731571)

In [27]:
1 - binom.cdf(9,15,0.7) # Answer of C

np.float64(0.7216214402043639)

In [28]:
binom.pmf(10,15,0.7) +  binom.pmf(11,15,0.7) + binom.pmf(12,15,0.7) + binom.pmf(13,15,0.7) +binom.pmf(14,15,0.7) + binom.pmf(15,15,0.7)

np.float64(0.7216214402043644)

So when you take a square root of p square, you are going to get up in Is that okay? I saw population variance, you have to divide and sample variance, n minus once the divide Okay, now, let's not do it on Excel, let's try to visualize because I want to give your brain some exercise. If suppose all these numbers were equal to 6, or very very close to 6, not equal to just very very close, 5.9, 5.6.1, 5.9, 6.1, which has such numbers. So what will be the means? Six. Six. Six. What will be the means? Because numbers are very close to six. That means square. If deviation is.1, square.1, you'll let it. Now, this is my statement very quickly. I say that as the numbers become more and more consistent, they are equal to each other, they are equal to me, the value of variance and standard deviation will go down. Yes. I repeat my statement, because the important statement is that as the numbers become equal to each other or close to equal to each other, lesser and lesser variation, more and more the value of variance and standard deviation will come down. Yes. As the variations decrease, the value of standard deviation and variance also decrease. decrease. As the variations in the data decrease, the value of variance and standard deviation also decrease. Got this point? How is the product in the concept of the 如果我一位在这里面的设计, Now what will be my question? Let me give you the question and then we will tell you. So suppose I have a data from Mumbai and I have a data from Bangalore. Okay? And right now I am telling you that as far as I took a sample of some people from Mumbai and I asked them what is their salary, took a sample of people from Bangalore, what is their salary. The standard deviation in Mumbai is coming out to be 40,000 and standard deviation in Bangalore is coming out to be 20,000. Every salary in Mumbai let's say is 80,000 and the in Bangalore is 1,20,000. So, all your conclusions now, cool question. So one right answer that I have got from Shriyas. Thank you. What do you understand from that? So, if I am giving you data, what will you draw? So by looking at the value of standard deviation means that the variations in the salary are more in Mumbai as compared to bankroll. So what do I have to say? I have to say, answer after we will put it here. Whoever has given this answer is perfectly fine. But my advice is add the bond. General economist to the point of which I can say. Disparity in the salaries is more in Mumbai as compared to bankroll. Disparity means? Variations. That means, rich is very rich in Mumbai and poor is very poor in Mumbai as compared to Bangla. There are two bagas which are relatively low as compared to Mumbai. I saw that. Sorry? What is average? What can you draw from the average? So average salary in Bangalore is more as compared to Mumbai. In bank account you get paid more as 100 to move by and not only you get paid more, the variation in the salary is all over the place. Got the point? I'm going to refill it. Thank you. So the question here is A, B and C are the key investment portfolios that you have. Okay? A is a digital FD, giving you a return of around 6.8%. FD per standard FD come over here? Come. Come over here. Fixed. Fixed. Fixed. So, what do you think about the standard deviation So, you get a return of 14% on if you put your money into BSE. 14% but variation is a lot. So, 4% is a standard deviation. So, what can happen? Something like mutual fund? Very good. Yeah, balance advantage from this point. Most of the returns would have come. So, if you have 4 columns data set, you create a 5th column based on these 4 columns. That is column feature in the region. You are creating new features, based on the existing one. Maybe that 5th feature can be used to many models. Now, I don't need to put it in standardize. It is the concept of the fact that I know. Now, I will create a new term. So, what I mean is So, 0.1 divided by 6.8, two values and whatever it is. If I just get a measure, I just throw. Somebody can just throw it on Excel, please. 0.1 divided by 6.8. 0.1631, I think. Is that right? 0.015. 0.015. . Now these values that we have got, these are the coefficient of variation. So what would we prefer? We would prefer that sigma should be as low as possible. That is what we would prefer with predictability actually. Somebody said about predictability. So the ideal investment would be lowest sigma and highest V. That is what we would prefer. Right? That means, now in these three things, I have said that, sir, the return is less than b. I am saying that the risk is less than b. Sir, the risk is less than b. Sir, the return is less than b. We have to balance both of them. So, what do we do? We have a coefficient of variation. Sigma divided by mu. Now, how much we can So, these are just the ways by which you are performing your analysis. This is one method. This is called coefficient of variation. And some books they multiply by 100 and put that in form of percentage, that's okay. Fraction linear percentage is a. So please make a note, this is called as coefficient of variation. It's a derived quantity, it's not a primary quantity. It's a derived quantity. Okay, now what I'll do is, I'm just putting this on my spider. 43657, 43657, 9487, 9487. Ye values such as not note kalli lega. You have noted all these values na? Hani ye values such as not kalli lega. Excel sheet markup 1, 2, 2, 1. Just note on these values for my reference. Just for my reference, note on these values. So now let me go to spiders. Now if I want to calculate the population variance first, then the command will be np.var and put x2. Your population variance comes out to be 4.04. Right? right? NP dot var gives you the population variance same way NP dot STD gives you population standard deviation which is 2.01. Now if you want to calculate sample variance because most of the data we have mentally you are going to do any kind of data analysis, it is going to be on a sample data. You hardly get to work on a population data or by default consider any data as a sample data unless until it has been explicitly mentioned that this is a population data, in your case That means these values of variance and stratification mathematically are not correct for me. Because these are population values, I want the sample value. So, scale carbon number, to get the sample variance function remains the same, you will have to add another attribute to it saying d d of equal to 1. When you put this d d of equal to 1 of this value of 4.48, you can just set up this your sample variance. And the same thing in your standard deviation, you have to write d d o f equal to 1, you you have got 2.12 which is your sample standard deviation. Thank you. I am going to study the first Now I have got two weighing scales. One weighing scale for measuring the weight of a man, man or woman, whatever. And the other one weighing scale is for gold. On an average, the weighing scale you will use now. The average weight which I measure is around 65 kg. And the standard The variation for that weighing scale is 3kg. It is already zero. The variation is 3kg. Gold may normally have to weigh a gold weight of 3.5 grams. And the standard variation is 5 grams. What machine ka variation is more significant? 3kg. Both. Dollars Because this also will have an application going forward. Because? Okay, quite, quite, quite. When you have to do this, this application will be very serious. And this is what happens to me. My job is to get an average student and a top performance. Average student knows the functions. Average student gets taken away from this question. and this essence is not a serious issue. I don't know. Be happy, I am not saying you are such a young girl. Be happy, be proud of your achievement. Yes sir, yes I've worked hard and I've worked 75. But sir, that's just a small milestone for me. I have a job now, sir, I have a job. Then dance. What should I do? Actually. I have done a lot of work. If you want a job, please tell me what you That is something. Why do you dance? Whoever works dancing, what do you do? I don't know what you do. But that just shows the attitude. That is the way you are happy. But that is the way you dance. Questions are coming to you. I suppose you have to understand the essence. Because that essence is extremely important. So, if you My question is, if positive and negative and zero were at and I wanted to avoid that, I could have taken absolute value also. Absolute value is minus 0, plus 1 plus 1 plus 1 plus 1 plus 1 plus 1 plus 1 plus 1 plus 1 plus 1 plus 1 plus 1 plus 1 plus 1 plus So, I will discuss this question and the answer in the class in the coming days. And this could be a question in your interview. Question is not a direct question, but you ask the question. To check your understanding of the subject, not the cramming power of the subject. How are you? Question is not a question. I will discuss this question with you when the time comes. But what I am saying is, abhi you aapso konsa kata hongga, isko jyaan ta sunna. As of now, to dho moh chiza bates to hain kam se kaam. This is also M, this is also M. But units are different. Can you compare two numbers with different units? No. Can you, suppose, wo yaan nahi ho ra, wo abhi aatikallte li will be shudikin hogi. Which of them is bigger? 5 kgs or 5 meters? Can't compare. Or aaj aapri vataanam, aaj hi galti karoge. Aaj bataanam aur galti karoge. Nab galti karoge na, prissiden ko yastara hao. 5 kgs and 5 meters cannot be compared with each other, but aap compared karoge. So, what is the So, sigma upon mu ki ki unit? Unit less, dimension less. So, Cs can be compared, but this was distance and this was weight. You cannot compare weight with distance, but when you calculate Cv, both of them become quantity then the comparison is possible. That is the importance of CV. Because CV is a dimensionless quantity so comparison across different industries, across different processes becomes possible. Here CV is important. What is this point? So just for the example say Suppose I take 3.5 km/h and take this to the km is greater than 5 but I cannot say that 7 meters is greater than 5 meters. This is a very important concept that this application again regression is coming from the PM. I will tell you that at least 50-60% of you will make this mistake of comparing the oranges with apples. So what you have done? We have done the range IQR, valiant, standard X, Excel calculation and your spiking calculation. Now, let's go to skewness and photopsis. I will put it on the box. This is curve A or graph A. Center one is B. And the one on the right side is C. Now let's talk about this graph A first. Which side is the peak? First of all, is it a symmetrical graph? A? No. So which side is the peak? Left side. Left side. So it's a limit. So, peak on the left side. Okay, when the peak is on the left side, so which values are having a high frequency? Smaller values have a high frequency or larger values have a high frequency? Smaller values. Smaller values have a high frequency. Anybody with an issue here? No. Okay, smaller values are more as compared to the bigger values. So, when the peak is on the left side, we say that the graph or the distribution is skewed to right, not left, right. Peek left, skewed to right. Or we can say it is positively skewed. And the skewness factor for this kind of a graph, the skewness is going to be greater than C. So again I think skewness kaise nikal ti hai, ka function kya hai, wo important na hi. But agar skewness positive hai, so tumara interpretation kya hooga, yeh important So, sci- is on the left side that means smaller values have a high frequency as compared to the bigger values. You got more of smaller values in your data set as compared to the bigger values. Again let me check about this practical example. Where in a real world would you see that smaller and bigger values are less. What are you? Perfect. Salary distribution has a huge amount. Large number of people would be getting a salary of something like 20-40 thousand per month. And how many people would be getting a salary of 3 lakh rupees per month? Very few. That frequency, number of people getting that salary of 3 lakh per month is very less. Very less. Y axis is frequency, X axis is the salary. You can see that the frequency is very small for high salaries. The frequency is very high for smaller salaries. Salary distribution in any company will follow this law. That's what I showed in the morning. The age that we have been in Mumbai, that's the graph of the company. So, we have lots of youngsters, very few senior citizens. Because we are living in an area where the IT sector is. Possibly that could be the same case with Mumbai. Relatively younger crowd here. Maybe I don't understand. But I think that's the same thing. I am not sure that the distribution is right. transactions build most category maa sir. So the value of UPI transactions will be this one. So, it is hope that this peak is on the left side. Now, the second one, the graph B, this is where peak is in the center. So distribution is symmetric. So this is skewness value will be equal to 0. For a symmetric distribution, the value of skewness will be equal to 0. Iska example mani gokou diya, sir. Any operation which you are doing multiple times, how much time you take to brush your teeth, how much time you take to get ready for the day, How much time to take to have the lunch? Some repeated common. All of these are repeated operations. They will follow this symmetric distribution called as a normal distribution, which is going to be your daily bread and butter for the next week 10 days. This distribution in the center, this is called as a normal bell-shaped curve. And the last one is your C. For the C I can write that peak is on the right side. So, it is not that distribution is skewed to the left negative is cube and the spinous factor here will be 0. ¿Qué es la mayoría de los estudiantes? Let us say that some 70-80% of you have got which are marks which are more than 70, 70-75% of the marks side. Now as a data analyst, okay? As a data analyst, is giving this information good in a pursuit or is there any better way of giving this information which is actually going to give you the real picture? I said that most of you have got high marks, 70-80% of those scenarios, or paper easy, or intelligent. And I agree with him. So that means, We are giving companies these numbers, that please come to my campus because every person of my students have got marks around 75 or above 70. From a data analyst part of you, what could be a better way of representing the situation, which could be more realistic. I have already taught you that concept. That concept has already been taught to you. What is wrong in reporting percentage? And what would be the better way of reporting your own performance? What would be the better way? Which would give you a more realistic picture. Percentile. Perfect, percentile. Because percentile will tell me, with respect to the others, where do you stand? So quite possible that everybody in India is getting 85 and above. You are 75% marks. Because your percentile is only going to be 20 percentile. Only 20 percent of people are below me. You might have still got 75 percent marks. So your percentile matters more than the percentage because percentile will tell me where do you stand with respect to the others in C-JAC and the other C-JACs also. That would be the real indicator of your performance. Because companies can't go to CREAC That means, we don't 90, then you are useless. In the sense, we don't use this. But in general, we are useless. 90 is not a matter of money. Because when all the hundred are, all the 90s are up to, so you are useless. Because the premium colleges, the premium seats, So, don't look at your percentage, look at your percent type. See, this is the stats of aahko kamna aahe. This is how you have to think. This is the way a data scientist has to think. When you go to your client, your client gives you the percentage, tell me the percent type. Aah apne bachche ko aah kohane hoongay, If we go to college, say, "Sir, average salary, not a lady, sir, tell me the median salary, don't tell me the mean salary." I know there is a difference between median and median. If you give me the mean, then, tell me the median salary. That is how data analysts are doing. Do you know how to play with numbers? Do you know how to play with numbers? This is what makes you a data scientist. Not cramming some concepts here and there. That's what I want, I hope by the end of these six weeks of training, I teach you a new language. That language of numbers. And essentially your entire thought process has not got changed. This is what means to be a data scientist, that you know how to deal with numbers, you know how to play around with the numbers. And you also know how the others are playing around with the numbers and you can catch them. Don't tell me the means, sir. So, this is the median salary. So, I decide that you can improve the population. 96 kids of the year, I think, so say 97, he did a great effort. But the bottom 20 percent effort. So his 97 is not interested. Maybe this percentage is 42, but that was a very, very difficult J.E. exam. So the 42 percent will be offered. Even after getting 42 percent, you might be at 99 percent higher. Because only 1 percent people got more than 42. That is the real indicator of the problem. So that is, this is what I am telling you, this is how your thought process should be as a data scientist, not just doing a course, learning some commands. Unless until your thought process changes, . So, both practice. Look out for the numbers. Where do I get numbers? I can't tell you how many Both of them are having an almost similar approach. One is the one. The one is the 51, the one is the 50, or the 52. But if you want the Batman to play for your life, I will go 50 a century, then you will go. Who would you like to play for your life? Sehwan or Dravid? Dravid. Why are they? Thank you. Long. What will happen to you? Variance is standard deviation. So, if you have a standard deviation, you can calculate it. When you have a standard deviation, you can give average, strike rate, and you can give standard deviation. But you know that reliability of a bad student is very important. You still don't give standard deviation. That's your stability. You do not know how to measure what you want. You need a bad student, but you don't know that So all the innings of Rahul Ravid calculated the standard deviation will be much lower than Sibidhul Seba. He never made a triple century, but he never got out on a zero also. That's consistency. But how do you know whether he's consistent or not? Because you need his value of standard deviation. Do you have it on 3 and 4? No, you don't. Do they ever show the value of standard deviation? No, they don't. That's the lack of knowledge about that. You know reliability is important, but you don't know how to measure it. This is a mistake. Not your mind. Not the point. When you go to the industry, you can pick up these kinds of points. These are the things keeping at the back of your mind. I will say this So, this is the reason why the concept is called it Sigma. Whole focus is on standard deviation, how to reduce standard deviation. So, when I go to industry and go to training, I don't know what to do. Sigma means standard deviation. So, my standard deviation is not. I will tell you how to do standard deviation. I will tell you how to do it. Because they do not have the knowledge of status. This is what we need to do. How to think in some process. And statistics will help you in doing that. That entire thought process that you have to do, that will help you in taking decisions based on your thoughts. This is where it comes into your thoughts. So, this is your QNES. I will make some examples. So, we will take a graph and then show the value. That will be a better done. So, let's do the right skew there and make some graphs quickly. So, just give me some time. Now let's take a step up. is probability distributions. There are five distributions that I need to do with you. Let me repeat, I am in other six, but I am in other six. Then make a list of those five. The next topic is probability distributions. So there are two distributions that I will be doing with you for the discrete data. So, first, write down distributions for discrete data. Discrete data counts on time. Which we can count. Counts. Counts. So, decimal points? Nine. Nine. Counts on time. So, discrete data can use. The two distributions are binomial, binomial distribution. And the second one is the Poisson, P O I S S O N, P O I S S O N, double S, Poisson, bimamal and Poisson. And continuous continuous distribution, exponential, exponential distribution, normal distribution, exponential normal and students t students t distribution f or shy square d So five institutions are participating in the name. First one is PANOLIN, Students t. I guess you can search. only t. I'll just give you one minute. Thank you. So the first one that we are going to do is the binomial distribution. So, the first distribution that we are going to do is the binomial distribution. So, I am going to see the rest of the distribution element right now. when there are only two possible occurrences. Binomial distribution is used when there are only two possible occurrences. That is why the name is zi binomial. Bi means two. Binomial distribution is used when there are only two possible occurrences and the probability of occurrence is known. The historical probability of occurrence is known. The first point that only two possible occurrences are there. Either I'll come for the class or I'll not come for the class. There's no doubt. And probability that I come for the class is known to you. or maybe you will tell you, "Khahan Kar Sathe Bola I'm going to bite him. Thank you. Okay, so this is the question that we are going to work on. Chances that it rains on any given day in the month of June in Mumbai is 0.7. 70% chance that it rains on any given day. Sorry? So, if I am in Mumbai for 15 days, so now you have to find out probability for these three events. What are the chances? Number one, out of those 15 days, it rains for exactly 10 days. Second, it rates for maximum 10 days, not more than 10 or minimum 10 days. So, these are the three cases of probability that we have to calculate. It's key to manual calculation for here. If I wrote you Python and I used it with a pen and a paper and with a character to me read. Anybody remember this? What is the formula? N - 1. but the right to And please keep noting down these import commands in that list which I asked you to make on the first page so that I don't have to tell them again and again. You have already done import scipy then from scipy import stax then from scipy.stax import Now, in these kind of cases in these all probability things that you are going to do, x is all I have to think about it more. Advanced analytics of AML exam. These questions came in the exam for all these distributions because it is part of your course content. Unfortunately, you are not using Python here. Even a man-po-pond-doctor is a teacher, I said: " So, we have to So, this was a shocker for me some time back, some three four batches back. I didn't know that this was a problem for me some time back, some students told me. So, that was a problem for me some time back, some students told me. Das ist das, dass es die Begriffe ist. So, the first one, binom.cdf, you have better not to PMF, first is emf, second is cdf, and the third one is sf. These are three options that you can use with your binom.f function. So, by non-dot you have to put a PMF or a CDF or a S and then give your less of importance. PMF for full form is probability mass function. The full form of PMF is probability mass function. PMF is going to be used when you want to find out the probability for an exact event. I want the rate to happen on exactly 10 out of 15 days, then that case number 1, you are going to use PMF. So, just write down, PMF is used to find out the probability of a exact number, exactly equal to something, not up to a minimum. PMF is used for finding out probability or an exact number. The full form of CDF litigate cumulative distribution function. CDF stands for cumulative distribution function. It is not required from the practical point of view. Practically, it is used. So, CDF is cumulative distribution function. It stands for cumulative distribution function. When you want to find out the probability, CDF will be used. When you want to find out the probability of up to a particular number. When you want to find out the probability of up to U, P, E, O, up to a particular number. So if I am saying that it rains for exactly 10 days, another 10 days, Barish Ogi. And you have those 10 out of 15 days, Barish Ogi. If I say up to 10 days, what are the numbers included for this same example? If I say up to 10, up to 5, up to 5, how are the numbers included? 0, 3, 4, 4, 4, up to 5, 5 included all right? 0 1 2 3 4 5. 0. 0. 0. It may not have been at all. That is up to 5. So what are the cases up to 5? 0 1 2 3 4 5. Yeah. So up to 5 up to 10. So, if I have to find out the probability for up to 10 days of rain, we will be using the 3 years. Yes,. A third part, it rains for minimum 10 days. So, it was not 2 1 minute, many of our sources. So, now if I want to find out the probability that it rains for exactly 10 days, I just binom.tmf. Now I'll just write that function here binom.tmf. Now the inputs that we have to give here, the first input will be for what number do we want to find out the probability. So that number will be 10. You want to find out the probability for 10. Out of how many days? 15. Second input is out of comment. So 10 days ki probability ni kaan ni in. 10 days aap Mumbai me ho ke tizi. 15 days. So that is the second output. And the third would be the event probability. What are the chances that it rains on any given day? That we took it as? I think 0.7. So this answer will be 0.206. Roughly 21% chance that it's going to rain for exactly 10 out of 15 days. Thank you. 2 divided by 3. probability that it raised on any given day. 0.0. Even probability. 0.0. This question is not the case. Just a moment. Now if I want to calculate the second question, the second question says that finds a probability that it raised for maximum 10 days or up to 10 days. So, here in place of Tmf, we will use CdF. So, that probably will be 0.48. So question number two is very important. I said what is the probability that it rains up to 10 days. So my first question was what are the numbers which are going to be muted here? Zero. Zero. One, two, ten. One, two, three, four, five. Ten-thakha. Ten-thakha. Ten-thak If you have to find out the answer to the second question, use the PMF function. PMF kya betta? Exact number is any property. So ap kya betta? So, what do you At 2. Because it will rain for either 0 days or 1 or 2 or 3 or 4. English for all right. And when all goes on, the probability is higher. Multiply how do we do? And. Now, what is the probability that it rains for minimum 10 days? Now, how many numbers are included? Name, zero name, zero name, 10. Name, 10, 11, 12, 13, 14, 15. Case number 3, the numbers are included are 10, 11 12, 13, 14, 15. Now, if the binomial monopoly twice account will be used when there are only two possibilities. question, that's a point. If chances that I come for the class is point 7, what are the chances that I don't come for the class? Point 3. Point 3. Point 3, how do I make it? Point 3. . . If you have a question, you will be asked to write the answer. If you have a question, you will be asked to write the answer. Repeat, answer is, this is the answer that I am giving you. Is answer question, what do you think? 是的 I know that you know. So, I am going to I have a question. Same. Yes. So neither has given me the right answer. Very good. My question, abhi may I have a question repeat. Yeh ki third ya command number 40 pey joe may ne probability nitaali hai yeh kis cheez ki probability. What is the event for which this is a probability? Or what is the question for which this is a probability? Sir. 0 to 10 10 to 15. Minimum 11. This is the probability for minimum 10 days of rain. Sorry, minimum 11 days. Minimum 11, sorry. My mistake. This probability, this expression that I have written on the command number 40 is the probability for minimum 11 days. When minimum question comes, binom.cdf bracket will be less than one. Be a data scientist, don't ratify this. Understand the logic, why am I doing that? Because when you put binoc.cdf, it includes 10. When you say up to 10, it includes 10. And I say minimum 10, where 10 included, right? If you want to include 10, then you want to include higher side, what should you do in lower side? 11. Now tell me the answer. This is my question number C. Now tell me the answer to the C part now. Now who will tell me the answer to the part number C? What should be my expression? What should I write here so that it becomes the answer to the part number C? 9, whatever. That would be the probability that it rains for 10 or more than 10 days. So, So, this is the answer to part C. Now, let's take a look at someone's doubts. So, 90% of the students, if I am asking them this part number C question, they will give me this as the answer. 90% of the students. 10% of the students, so this is the correct answer. Command number 41 is correct. Now I show that to you. What are the days which are okay with us? Video 10, 11, 12, 13, 14, 15. Now this way. 10, 11, 12, 13, 14, 15. 12 13 14 15 So, let's go True, have 2 minutes left? or it was not absorbed. I will repeat everything that I have said, but you first try to absorb it whatever you want. Take two minutes, I am now. I am going to keep quiet for two minutes. See what we have done on the board. Then I will recap it first. What is the manual? What is the manual? How do you do this? The N What's up? What's up? I'm going to go to the He trains for minimum 10 days. - He trains for minimum 10 days. - I don't know, I'm a kid. - I see, English, - I'm a kid. - Deneshrim. - I'm a kid. - Ask related to Dena. - Take related? - Dena'shrim. - Just ask. - Dyeodad. So, I hope command number 38 is clear to everyone? Yes. Command number 38 main, the option that we are using is PMF. PMF function is used when you want to calculate the probability for an exact number. So, if you see number in probability in that case because for discrete data is that possible? Yes, discrete data is possible that I can come here for exactly 7 days it is possible. I am not sure Thank you. Okay, let's take a quick recap please. The command number 38 where I'm using the PMF function, the PMF is going to be used when you want, might I have stamps for us please? So, PMF would be used when you want to calculate the profitability for an exact number that means equal to 10, not less than equal to or greater than equal to, please mind that, it is equal to 10. So, for equal to 10, we use PMF, transfer to 0.206 RL that means 20.6 percent chance that it range for exactly 10 out of 15 days. Now, what are those 10 days? What should be 10 days? First time was 10, last time was 10, if you clean for those in the UF, clean and going put those 10. Yeah? Now, let us move on to the command number 39. 39 says CDF which I said will be used when you want to find out the probability of up to a particular value. That means when you are adding 10 here as an input to serial, probabilities will be calculated for all numbers less than equal to 10. That is why the numbers included here. Zero say 10. Zero has to be done. I have to say that it is not a bad thing. At least this is serial. Zero. Last six days, how many days are waiting? Zero. So, 0 will be included from 0 up to 10 that means I can say less than equal to 10 all less than equal to 10 will get included in CDF. Now the third question said, now third question is the trick. PMF has the exact number CDF, up to that number starting from 0. When the question said what is the probability that it will range for minimum 10 days that means question says what is the probability that it range for greater than equal to 10 days. So equal to be include. I lose the probability of equal to 10. So how to make sure that the 10 comes on the 1 minus assignment, put 9 there. So 10 up by 100, that's what you want. Because I said minimum 10 days, greater than equal to 10 days. So 1 minus series 9, so that 9 up, 1 minus 9 will be, sorry 1 minus 1 dot cdf 9 would be starting from 10 and further. Got this point? I will answer your question. Yes. Yes. Yes. I said greater than equal to 10. So, 10, 11, 12, 13, 14, 15, 16. If I add this, it will be 0.7. But when I was 10 from the left hand, I was 0.5. What I missed? I missed the property. If I had 10 here, this property goes on the other side. But if I had 10 here, then what would I have to do? Then up now, that's what I got here. Keep 9 on the other side. Subtractor do. Toh, which you accept that will start from 10. Because rejected is up to 9. At this point, I will provide an eye owner, which will be useful for you. The statistics of machine learning, which portion is mine. In-depth statistics, PML, machine learning, regulation, regulation, all these things are just a subset of Six Sigma. I am going to on Six Sigma. Now see, you can tell me Six Sigma. It's one of the vibles of Six Sigma. If you want to learn something, you can take it. In this book, what are the skill sets that a person should have as a Six Sigma specialist? I want to show that skill set. And the ability of importance it has. skill number one is do you have the courage to overcome obstacles completely related to your attitude which is also your second point attitude third point logical thought process then comes your communication skills which i talked about You know what? This is the industry requirement. Think about it. I don't want to learn math. I'm not going to go to the math. I'm not going to go to the math. Many things are important. Logical mindset is my own. You can think about numbers, then math is coming. If you don't know math, I'm not going to learn 4 things. I am not sure what I important, there is no doubt about it. But I cannot take it. Your attitude is something which you are there with. After this course too, very very important. These two are extremely extremely important. I could not because of and I could in spite of. That is overcoming obstacles. That change in the mindset. That's overcoming obstacles. I will discuss this. I will discuss this. Losers will always say I could not because of this. Sir, I have a problem with my wife, my father has not given money, my mother has also given her brother, that's why I have done it. Overcoming options are not to give money, I have to give money, I have to give money, but I have to give money. But I have to give money. That's overcoming options. This is the way you This will show you how important this logical thinking is, how important it is for you to be data driven. I have to say So, what are the soft skills that I need as a data scientist, as an ML engineer? Hard skills are technical skills. Who will teach me? Who will teach me? What are the soft skills that I need as a data scientist, as an ML engineer? Because companies will not only check for your heart skills or technical skills. They will not check your attitude. That's what happens. My experience is that I will tell you what you are in the middle of 10 months. I will be sitting in the middle of the experience. Your body language will be very important. You are the same. When you are the same, where are your hands? 你做不错 I don't know how to read the body language. I am going to

04:30


In [30]:
from scipy.stats import norm,t

In [31]:
norm.cdf(14,16,2.3)

np.float64(0.19226902649831168)

In [32]:
1 - norm.cdf(14,16,2.3)

np.float64(0.8077309735016883)

In [33]:
norm.sf(14,16,2.3)

np.float64(0.8077309735016883)

In [36]:
def get_full_name(first_name, last_name):
    full_name = first_name.title() + " " + last_name.title()
    return full_name


print(get_full_name("gajendra", "rathod"))

Gajendra Rathod


In [39]:
norm.cdf(14,16,2.3)

np.float64(0.19226902649831168)

In [37]:
z = (14-16)/2.3



In [38]:
norm.cdf(z)

np.float64(0.19226902649831168)

In [40]:
norm.sf(z)

np.float64(0.8077309735016883)

In [41]:
z

-0.8695652173913044